### ISSUE TITLE
Update RDF and replace AFV-EuroScope Bridge plugins

### ISSUE THREAD

**Issue by luke11brown** [VATSIM-UK/uk-controller-pack#686]
## What is the current situation?
AFV bridge and RDF plugins are versions included:

- https://github.com/AndyTWF/afv-euroscope-bridge
- https://github.com/chembergj/RDF

## What is the proposed change?
Replace both of these plugins with the single solution provided:

- https://github.com/KingfuChan/RDF

## Additional context

This provides the following benefits:

- Single plugin provides the functionality of the previous two
- AFV bridge compatible with other voice software such as VectorAudio & TrackAudio
- RDF settings can be defined on a per ASR basis
- More RDF filtering/customisations available

**Comment by kristiankunc** [comment]
I'd like to do this.
Is it just a matter of replacing the plugin DLLs or some additional configuration should be done?

**Comment by luke11brown** [comment]
Certainly replacing the DLLs is a start.

I think we'll be looking to also do the following:

- Set precision values
- Set altitude filters for both precision and total functionality.

This can be achieved via the settings described https://github.com/KingfuChan/RDF/blob/master/README.md

For now, please set the same values for all ASRs and we can discuss what is suitable @hazzas-99 @PLM1995

**Comment by kristiankunc** [comment]
I've tried playing around with the plugin and it ended up crashing my ES quite a lot. I've created an issue on the RDF repo to track it. https://github.com/KingfuChan/RDF/issues/5

**Comment by PLM1995** [comment]
> I've tried playing around with the plugin and it ended up crashing my ES quite a lot. I've created an issue on the RDF repo to track it. [KingfuChan/RDF#5](https://github.com/KingfuChan/RDF/issues/5)

I'm assuming you've already followed the steps to mitigate against the known issues which cause ES crashes, but since your report is rather minimalist, might be worth clarifying what other plugins were you using at the time, and how you had them configured.

**Comment by kristiankunc** [comment]
> I'm assuming you've already followed the steps to mitigate against the known issues which cause ES crashes, but since your report is rather minimalist, might be worth clarifying what other plugins were you using at the time, and how you had them configured.

Yeah I have, not sure what caused it possibly some odd config but seems to work alright now

**Comment by luke11brown** [comment]
Ok,

so we'll need to set up all Plugins.txt with the settings below

PLUGINS
```
<eventually existing configuration lines>
RDF Plugin for Euroscope:VectorAudioAddress:127.0.0.1:49080
RDF Plugin for Euroscope:VectorAudioTimeout:300
RDF Plugin for Euroscope:VectorAudioPollInterval:200
RDF Plugin for Euroscope:VectorAudioRetryInterval:5
RDF Plugin for Euroscope:RGB:255:255:255
RDF Plugin for Euroscope:ConcurrentTransmissionRGB:255:0:0
RDF Plugin for Euroscope:Radius:20
RDF Plugin for Euroscope:Threshold:-1
RDF Plugin for Euroscope:Precision:0
RDF Plugin for Euroscope:LowAltitude:0
RDF Plugin for Euroscope:HighAltitude:0
RDF Plugin for Euroscope:LowPrecision:0
RDF Plugin for Euroscope:HighPrecision:0
RDF Plugin for Euroscope:DrawControllers:0
END
```

And we'll need to set up all ASRs with the ASR specific settings

I'm not one that actually uses this plugin, as it's unrealistic (in the UK) However, my understanding is it uses triangulation from transceivers, therefore it is only accurate at high levels, I suggest we effectively disable it at low levels, as ground controllers should not need to use this. If they are using it, in my opinion it shows poor management of their frequency. Of course, I'm happy to be corrected if someone else knows different on how it works/where it is used IRL.

ASR settings below
```
PLUGIN:RDF Plugin for Euroscope:RGB:255:255:255
PLUGIN:RDF Plugin for Euroscope:ConcurrentTransmissionRGB:255:255:0
PLUGIN:RDF Plugin for Euroscope:Radius:20
PLUGIN:RDF Plugin for Euroscope:Threshold:-1
PLUGIN:RDF Plugin for Euroscope:Precision:0
PLUGIN:RDF Plugin for Euroscope:LowAltitude:0
PLUGIN:RDF Plugin for Euroscope:HighAltitude:0
PLUGIN:RDF Plugin for Euroscope:LowPrecision:0
PLUGIN:RDF Plugin for Euroscope:HighPrecision:0
PLUGIN:RDF Plugin for Euroscope:DrawControllers:1
```

**Comment by kristiankunc** [comment]
> I suggest we effectively disable it at low levels, as ground controllers should not need to use this.

I see what you mean. I am not sure how many people actually use it on ground, but I do personally, even though it's unrealistic. So I wouldn't be for "disabling" it on ground, rather make it work as well for those that want it.

**Comment by PLM1995** [comment]
I fully agree with Luke, unless it's an IRL thing, there is absolutely no need for a GND/TWR controller to use RDF.

In general we go for the pack being realistic, then if people want to tweak things to make it unrealistic they can do that themselves @kristiankunc, I wouldn't be in favour of this being on by default for GND/TWR (unless as Luke says it's a RW thing).

**Comment by luke11brown** [comment]
Further thoughts,

If we adjust the precision to a realistic value for low altitudes then this will also make the use in ADC situations almost pointless, as  it will likely not be precise enough over the space of an airfield.

**Comment by kristiankunc** [comment]
> If we adjust the precision to a realistic value for low altitudes then this will also make the use in ADC situations almost pointless

> I wouldn't be in favour of this being on by default for GND/TWR (unless as Luke says it's a RW thing).

I am certainly for disabling it by default, but I believe if someone wants to use it as GND/TWR, they should be able to enable it with usable settings. Even though it is unrealistic.

**Comment by kristiankunc** [comment]
Okay I'd like to get back to this, do you think that leaving the functionality for ADC while disabling it by default is not desirable?

**Comment by luke11brown** [comment]
The main reason for this PR was to consolidate the RDF and AFV bridge plugins to a single plugin that is being actively developed, whilst adding the websocket access for TrackAudio.

Unfortunately, it appears that we can't have the AFV bridge functions added with this plugin, without adding unrealistic functions of RDF. IMO, the pack should aim to simulate the various ATM systems used by controllers in the UK. RDF isn't one of those things. That's why we include it at the moment, but don't load the plugin by default, as we understand some controllers like the RDF functionality.

I therefore think that instead, we should look to ask Andy to include the TrackAudio integration in the AFV bridge plugin. I will raise an issue there and leave this issue open for now.

**Comment by GeorgeBarlow** [comment]
> The main reason for this PR was to consolidate the RDF and AFV bridge plugins to a single plugin that is being actively developed, whilst adding the websocket access for TrackAudio.
> 
> Unfortunately, it appears that we can't have the AFV bridge functions added with this plugin, without adding unrealistic functions of RDF. IMO, the pack should aim to simulate the various ATM systems used by controllers in the UK. RDF isn't one of those things. That's why we include it at the moment, but don't load the plugin by default, as we understand some controllers like the RDF functionality.
> 
> I therefore think that instead, we should look to ask Andy to include the TrackAudio integration in the AFV bridge plugin. I will raise an issue there and leave this issue open for now.

I'm just thinking in terms of getting something done sooner rather than later regarding this, is it not worth reaching out to KingfuChang (RDF maintainer) and asking if he would render on his display type (as vSMR does) so we can only toggle the rendered circles on non-ADC positions (`RDF_MODE=1` in TopSky) which wouldn't require a whole re-implementation of the TrackAudio SDK in AFV bridge. RDF is also actively maintained and has the older AFV client implementation.

**Comment by kristiankunc** [comment]
I don't know that much about the RDF implementation but I thought that we could always just prevent it from drawing on SMR/Radar screens whilst keeping them loaded.

**Comment by GeorgeBarlow** [comment]
> I don't know that much about the RDF implementation but I thought that we could always just prevent it from drawing on SMR/Radar screens whilst keeping the plugin loaded.

You need to ensure it is its own display type as I believe EuroScope will default all drawing (that has not been otherwise moved onto its own display) to the main Standard ES Screen which we wouldn't want to remove.  If it had its own display type such as `RDF Display` it would allow us to toggle it depending on the ASR. I'm not familiar with the exact details of the RDF implementation in TopSky and if it does some things better but from the looks of things the standalone RDF plugin does a pretty good job at providing a lot of customisation whilst handling all this other TrackAudio and AFV client stuff, so there wouldn't be a need for AFV bridge.
